### DISCUSSION TITLE
How to generate a WDM driver?

### DISCUSSION THREAD

**Discussion by willingbug** [microsoft/windows-drivers-rs#32] [Q&A] [answered]
How to generate a WDM driver?

**Comment by Patuti** [comment]
This is how I did it:
```rust
// build.rs
fn main() -> Result<(), wdk_build::ConfigError> {
    let mut config = wdk_build::Config::from_env_auto()?;
    config.driver_config = wdk_build::DriverConfig::WDM();
    config.configure_binary_build();
    Ok(())
}
```

**Reply by wmmc88** [reply]
This works as a workaround to set the correct linker settings for WDM drivers. 

Note however that the bindings generated in `wdk-sys` will still include apis from the latest KMDF version. This should be fine. I am currently working on a way to configure `wdk-sys` to based off `wdk_build::Config`. Currently, this is only possible by modifying the config in the `build.rs` of `wdk-sys`, but I'm exploring options on being able to configure this via Cargo Features.

**Comment by jdoubleu** [comment, marked as answer]
Is this still an issue today, since there is `cargo wdk new --wdm my-wdm-driver` or can this discussion be resolved. I think the [workaround](https://github.com/microsoft/windows-drivers-rs/discussions/32#discussioncomment-7474313) by @Patuti isn't required anymore, because the driver type can be set in `Cargo.toml`  according to the [README](https://github.com/microsoft/windows-drivers-rs/blob/36558802149bc92455bf5719fe77f3a829d8580f/README.md#adding-windows-drivers-rs-to-your-driver-package).
